pandas/core/apply.py

```diff
@@ -628,7 +628,11 @@ def normalize_dictlike_arg(
 
             cols = Index(list(func.keys())).difference(obj.columns, sort=True)
             if len(cols) > 0:
-                raise KeyError(f"Column(s) {list(cols)} do not exist")
+                raise KeyError(
+                    f"Column(s) {list(cols)} do not exist, "
+                    "if you don't mean to apply to each rows, "
+                    "consider using `axis=0` instead."
+                )
```

Comment:
The problem is that the axis labels can be either columns or rows (indexes) depending on the given axis, but the error message always assumes the axis labels are columns.

Reply:
Would it be better to change it to:

```python
raise KeyError(
    f"Row(s) or Column(s) {list(cols)} do not exist")
```

If so, there are also some existing tests that match the error message that need to be updated. I will update them later.

Reply:
Yes, looks good.

I have not looked at this part of the code myself but if you can identify whether the labels are columns or rows that would be even better

Reply:
I belive that is actually what we want.

```python
df = pd.DataFrame({'foo': [2,4,6], 'bar': [1,2,3]})
df.index = ['A', 'B', 'C']
```

```python
df.transform(  
    {
        'A': lambda x: x+2,
        'B': lambda x: x*2
    },
    axis=0
)
# KeyError: "Row(s) or Column(s) ['A', 'B'] do not exist"

df.transform(  
    {
        'foo': lambda x: x+2,
        'bar': lambda x: x*2
    },
    axis=1
)
# KeyError: "Row(s) or Column(s) ['bar', 'foo'] do not exist"
```